libzfs_core/_libzfs_core.py

```diff
@@ -655,6 +660,276 @@ def lzc_exists(name):
     return bool(ret)
 
 
+def _uncommitted(check_func = None):
+    def _uncommitted_decorator(func, check_func = check_func):
+        if check_func is None:
+            check_func = func
+
+        @functools.wraps(func)
+        def _f(*args, **kwargs):
+            if not is_supported(check_func):
+                raise NotImplementedError(func.__name__)
+            return func(*args, **kwargs)
+        return _f
+    return _uncommitted_decorator
+
+
+@_uncommitted()
+def lzc_promote(name):
+    '''
+    Promotes the ZFS dataset.
+
+    :param str name: the name of the dataset to promote.
+    '''
+    ret = _lib.lzc_promote(name, _ffi.NULL)
+    xlate.lzc_promote_xlate_error(ret, name)
+
+
+@_uncommitted()
+def lzc_rename(source, target):
+    '''
+    Rename the ZFS dataset.
+
+    :param source name: the current name of the dataset to rename.
+    :param target name: the new name of the dataset.
+    '''
+    ret = _lib.lzc_rename(source, target, _ffi.NULL, _ffi.NULL)
+    xlate.lzc_rename_xlate_error(ret, source, target)
+
+
+@_uncommitted()
+def lzc_destroy_one(name):
+    '''
+    Destroy the ZFS dataset.
+
+    :param str name: the name of the dataset to destroy.
+    '''
+    ret = _lib.lzc_destroy_one(name, _ffi.NULL)
+    xlate.lzc_destroy_xlate_error(ret, name)
+
+
+lzc_destroy = lzc_destroy_one
+
+
+@_uncommitted()
+def lzc_inherit(name, prop):
+    '''
+    Inherit properties from a parent dataset of the given ZFS dataset.
+
+    :param str name: the name of the dataset.
+    :param str prop: the name of the property to inherit.
+    '''
+    ret = _lib.lzc_inherit(name, prop, _ffi.NULL)
+    xlate.lzc_inherit_prop_xlate_error(ret, name, prop)
+
+
+lzc_inherit_prop = lzc_inherit
+
+
+@_uncommitted()
+def lzc_set_props(name, prop, val):
+    '''
+    Set properties of the ZFS dataset.
+
+    :param str name: the name of the dataset.
+    :param str prop: the name of the property.
+    :param str val: the value of the property.
+    '''
+    props = { prop: val }
+    props_nv = nvlist_in(props)
+    ret = _lib.lzc_set_props(name, props_nv, _ffi.NULL, _ffi.NULL)
+    xlate.lzc_set_prop_xlate_error(ret, name, prop, val)
+
+
+lzc_set_prop = lzc_set_props
+
+
+@_uncommitted()
+def lzc_list(name, options):
+    (rfd, wfd) = os.pipe()
+    fcntl.fcntl(rfd, fcntl.F_SETFD, fcntl.FD_CLOEXEC)
+    fcntl.fcntl(wfd, fcntl.F_SETFD, fcntl.FD_CLOEXEC)
+    options['fd'] = int32_t(wfd)
```

Comment:
Mutating the dictionary passed in is usually not a good idea.  The caller might save or re-use the dictionary and be surprised at the change.